### PR TITLE
#3 Copied over the "warning.R" and "test-warning.R" from admiral repo…

### PR DESCRIPTION
**Pull request by PoojaKumari05**
…. No updates done.

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/main/articles/programming_strategy.html#deprecation-1)?
- [ ] Update to all relevant roxygen headers and examples.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/main/reference/index.html)" page.
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue so that it closes after successful merging.
- [ ] Address all merge conflicts and resolve appropriately.
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/warnings.R [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    warn(msg)
+  } else if (length(existing_vars) > 1L) {
+    msg <- paste("Variables", enumerate(existing_vars), "already exist in the dataset")
+    warn(msg)
+  } else {
+    invisible(NULL)
+  }
+}
+
+is_valid_dtc <- function(arg) {
```
Comment (bms63) [inline R/warnings.R:38]: #21 Can you update these functions missing roxygen headers please

File: R/warnings.R [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      "2003-12-15T-:15:18\n",
+      "2003-12-15T13:-:19\n",
+      "--12-15\n",
+      "-----T07:15"
+    )
+    warn(paste(main_msg, tbl, info, sep = "\n"))
+  }
+}
+
+warn_if_incomplete_dtc <- function(dtc, n) {
```
Comment (bms63) [inline R/warnings.R:107]: Update header

File: tests/testthat/test-warnings.R [hunk 3]
```diff
@@ -0,0 +1,44 @@
+library(admiral.test)
```
Comment (bms63) [inline tests/testthat/test-warnings.R:1]: Can you update these tests so they follow our unit test guidance for numbering.  
image

Comment (bms63) [reply tests/testthat/test-warnings.R:1]: You might be able to use the testthat adding that we created in admiraldev for just this purpose.

**Review by bms63 — CHANGES_REQUESTED** [review]
Please add roxygen headers and tests numbering

**Comment by PoojaKumari05** [comment]
> Please add roxygen headers and tests numbering

Hi @bms63 - Yes, I will update these now. The previous commit was just for copying over the files from admiral-core to admiraldev. I have not updated anything in the code yet but will start updating now. Thanks for the heads-up, noted all the points.  :)

**Comment by bms63** [comment]
Hey @PoojaKumari05 we resolved this in another PR so closing now.

**Pull request closed without merging**